npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -74,6 +75,7 @@ def initialize(dependency:, credentials:, dependency_files:,
         def latest_resolvable_version
           return latest_allowable_version if git_dependency?(dependency)
           return if part_of_tightly_locked_monorepo?
+          return if types_update_available?
```

Comment:
Please pardon my ignorance here 😇 Do we return early here because we have more than one dependency to return and will need to do a full unlock?

Reply:
Yes, at least I think :) I'm also mostly ignorant, but that seemed to be the pattern for peer deps and the "tightly locked" deps, so I followed it.  Would love input from @jurre on this.